bamboo/unit_tests/test_unit_layer_argmax.py

```diff
@@ -0,0 +1,232 @@
+    data_reader.python.num_samples_function = 'num_samples'
+    data_reader.python.sample_dims_function = 'sample_dims'
+
+    return message
+
+# ==============================================
+# Setup PyTest
+# ==============================================
+
+# Generate test name based on file name
+_test_name = os.path.splitext(os.path.basename(current_file))[0]
+
+# Primary test method
+def _test(cluster, executables, dir_name, compiler_name):
+    tools.process_executable(_test_name, compiler_name, executables)
+
+    # Import LBANN Python frontend
+    if compiler_name == 'exe':
+        exe = executables[compiler_name]
+        bin_dir = os.path.dirname(exe)
+        install_dir = os.path.dirname(bin_dir)
+        build_path = '{i}/lib/python3.7/site-packages'.format(i=install_dir)
+    else:
+        if compiler_name == 'clang6':
+            path = 'clang.Release'
+        elif compiler_name == 'clang6_debug':
+            path = 'clang.Debug'
+        elif compiler_name == 'gcc7':
+            path = 'gnu.Release'
+        elif compiler_name == 'clang6_debug':
+            path = 'gnu.Debug'
+        elif compiler_name == 'intel19':
+            path = 'intel.Release'
+        elif compiler_name == 'intel19_debug':
+            path = 'intel.Debug'
+        path = '{p}.{c}.llnl.gov'.format(p=path, c=cluster)
+        build_path = '{d}/build/{p}/install/lib/python3.7/site-packages'.format(
+            d=dir_name, p=path)
+    print('build_path={b}'.format(b=build_path))
+    sys.path.append(build_path)
+    import lbann
+    import lbann.contrib.lc.launcher
+
+    # Setup LBANN experiment
+    trainer, model, data_reader, optimizer = setup_experiment(lbann)
+
+    # Run LBANN experiment
+    kwargs = {
+        'account': 'guests',
+        'nodes': 1,
+        'partition': 'pbatch'
+    }
+    experiment_dir = '{d}/bamboo/unit_tests/experiments/{t}_{c}'.format(
+        d=dir_name, t=_test_name, c=compiler_name)
+    error_file_name = '{e}/err.log'.format(
+        e=experiment_dir, c=compiler_name)
+    return_code = lbann.contrib.lc.launcher.run(
+        trainer=trainer,
+        model=model,
+        data_reader=data_reader,
+        optimizer=optimizer,
+        experiment_dir=experiment_dir,
+        job_name='lbann_{}'.format(_test_name),
+        **kwargs)
+    tools.assert_success(return_code, error_file_name)
+
+# Construct methods that will be detected by PyTest
+def _test_clang6(cluster, exes, dirname):
+    _test(cluster, exes, dirname, 'clang6')
+def _test_gcc7(cluster, exes, dirname):
+    _test(cluster, exes, dirname, 'gcc7')
+def _test_intel19(cluster, exes, dirname):
+    _test(cluster, exes, dirname, 'intel19')
+def _test_exe(cluster, dirname, exe):
+    if exe is None:
+        e = 'test_{}_exe: Non-local testing'.format(_test_name)
+        print('Skip - ' + e)
+        pytest.skip(e)
+    exes = {'exe': exe}
+    _test(cluster, exes, dirname, 'exe')
+globals()['{}_clang6'.format(_test_name)] = _test_clang6
```

Comment:
Add comment above like `# Auto-generate test names by appending compiler to primary test name`